Makefile

```diff
@@ -404,7 +428,8 @@ e2e-helm-deploy:
 		--set linux.enabled=true \
 		--set syncSecret.enabled=true \
 		--set enableSecretRotation=true \
-		--set rotationPollInterval=30s
+		# Updated for mock provider testing. Mock provider returns default secret value as "rotated" once reconciler kicks in after 30s, but some e2e-provider tests takes more than 30s to complete before testing for rotation.
```

Comment:
Could you provide more info on this behavior? This is the flow:

1. `Mount` RPC call from driver to provider as part of `NodePublishVolume`. At this point, the pod isn't in the mock provider cache. So it'll add the pod to the cache and return "secret".
2. After 30s, rotation reconciler calls make a `Mount` RPC call to the provider. Now, the pod is already in the cache, so return "rotated".

So, why would this increase in timeout be required? With this change we're also increasing the other provider tests to wait for `150s` which is long?

Reply:
Step 2 intermittently occurs before we could run tests where we expect `secret` but gets `rotated`. Hence longer rotationPollInterval time to delay the rotation.

Initially I thought about increasing poll interval just for `e2e-provider` tests but implementation seems convoluted at the time given our make targets. But let me give it another try.

Reply:
I'm fine making this change in a follow-up PR. We can get this PR merged after the other 2 comments are resolved.

Reply:
Actually I reverted the change of polling interval and implemented more deterministic behavior for rotation test